### PR TITLE
Add example deprecated and default 2.0

### PR DESCRIPTION
**Pull request by xclebel**

### CODE REVIEW

File: swagger-ext/src/main/scala/org/scalatra/swagger/SwaggerCommand.scala [hunk 1]
```diff
@@ -63,7 +63,12 @@ object SwaggerCommandSupport {
 
   private[this] def modelFromCommand[T <: Command](cmd: T, fields: List[Parameter]) = {
     val modelFields = fields map { f =>
-      f.name -> ModelProperty(f.`type`, f.position, required = f.required, allowableValues = f.allowableValues)
+      f.name -> ModelProperty(
+        `type` = f.`type`,
+        position = f.position,
+        required = f.required,
+        default = f.defaultValue
+      )
```
Comment (dreisch-coveo) [inline swagger-ext/src/main/scala/org/scalatra/swagger/SwaggerCommand.scala]: does this change follow scalatra's formatting style?
whitespace changes are pretty significant when contributing to a 3rd party project

File: swagger-ext/src/test/scala/org/scalatra/swagger/SwaggerCommandSupportSpec.scala [hunk 2]
```diff
@@ -38,7 +38,6 @@ object SwaggerCommandSupportSpec {
     get("/new", endpoint("new"), operation(newOperation)) { "OK" }
 
     protected def applicationDescription: String = "The command support servlet"
-
```
Comment (dreisch-coveo) [inline swagger-ext/src/test/scala/org/scalatra/swagger/SwaggerCommandSupportSpec.scala]: whitespace consideration here too. I would avoid any cosmetic changes as much as possible. not just in hopes o merging upstream but to ease our efforts of keeping our fork up to date

File: swagger/src/main/java/org/scalatra/swagger/runtime/annotations/ApiModelProperty.java [hunk 3]
```diff
@@ -42,4 +42,14 @@
     * ordering, you should specify property order to keep models consistent across different VM implementations and versions.
     */
     int position() default 0;
+
+    /**
+     * Example value of the property. The format must be a valid json value strignyfied.
+     */
+    String example() default "";
```
Comment (dreisch-coveo) [inline swagger/src/main/java/org/scalatra/swagger/runtime/annotations/ApiModelProperty.java]: why would this be json?

File: swagger/src/main/scala/org/scalatra/swagger/Swagger.scala [hunk 4]
```diff
@@ -85,20 +85,31 @@ object Swagger {
...
-      allowableValues = convertToAllowableValues(allowableValues)
+      allowableValues = convertToAllowableValues(allowableValues),
+      example = example.flatMap(_.blankOption),
+      default = default.flatMap(_.blankOption)
     )
     //    if (descr.simpleName == "Pet") println("The property is: " + mp)
     prop.name -> mp
   }
   def modelToSwagger(klass: ScalaType): Option[Model] = {
+
```
Comment (dreisch-coveo) [inline swagger/src/main/scala/org/scalatra/swagger/Swagger.scala]: whitespace... i probably won't mention them all but please review

File: swagger/src/main/scala/org/scalatra/swagger/Swagger.scala [hunk 5]
```diff
@@ -109,15 +120,24 @@ object Swagger {
...
-          val asModelProperty = toModelProperty(descr, Some(annot.position()), annot.required(), annot.description().blankOption, annot.allowableValues())_
+          val asModelProperty = toModelProperty(
+            descr = descr,
+            position = Some(annot.position()),
+            required = annot.required(),
+            description = Option(annot.description()),
+            allowableValues = annot.allowableValues(),
+            example = Option(annot.example()),
+            default = Option(annot.defaultValue())
+          )_
```
Comment (dreisch-coveo) [inline swagger/src/main/scala/org/scalatra/swagger/Swagger.scala]: last mention for formatting changes too. even if this does match their style guide I would avoid to change unless it comes from upstream.

File: swagger/src/main/java/org/scalatra/swagger/runtime/annotations/ApiModelProperty.java (hunk 3 above)
Comment (xclebel) [reply swagger/src/main/java/org/scalatra/swagger/runtime/annotations/ApiModelProperty.java]: The best would have been to make this a `T` Type, but this is not a generic class, it's an annotation, and the type can't be anything. Any good Idea on how should I replace that type?

Note that in the end the json gets parsed printed directly in the swagger document, effectively allowing any type here.

File: swagger/src/main/scala/org/scalatra/swagger/Swagger.scala (hunk 4 above)
Comment (xclebel) [reply swagger/src/main/scala/org/scalatra/swagger/Swagger.scala]: Yes I was considering them but I'll do an extra effort

**Review by dreisch-coveo — APPROVED** [review]

**Review by malaporte — APPROVED** [review]

**Pull request merged**